sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -54,8 +55,11 @@ private[spark] object HiveUtils extends Logging {
     sc
   }
 
+  private val supportedHive2ShortVersions = Set("2.3.0")
```

Comment:
It seems `shortVersion` more safer that branch name:
![image](https://user-images.githubusercontent.com/5399861/55278947-d739fe00-534d-11e9-900d-9bc3269a90cc.png)

Reply:
Won't "2.3.4" or other 2.3.x versions fail to match "2.3.0"?

Reply:
How about just matching the prefix? `2.3`